unsafe_iface.go

```diff
@@ -57,8 +57,5 @@ func (type2 *UnsafeIFaceType) UnsafeIsNil(ptr unsafe.Pointer) bool {
 		return true
 	}
 	iface := (*iface)(ptr)
-	if iface.itab == nil {
-		return true
-	}
-	return false
+	return iface.itab == nil
```

Comment:
As you want to make the code shorter (I would not call that "optimization", as I expect that the compiler is able that change by itself), you can go further:

```suggestion
	return ptr == nil || (*iface)(ptr).itab == nil
```